doc/migrations/4.0.1_4.x.x.md

```diff
@@ -6,3 +6,46 @@ Currently, only the `urn:xmpp:http:upload:0` XMLNS is served.
 All major, modern client libraries and applications support the 0.3.0+ specification.
 If you experience any issues with making requests to the HTTP File Upload service, please update your client.
 
+## Retirement of old `*.cfg` format
+
+Since ____ release, we are no longer supporting `*.cfg` MongooseIM configuration format. For further application's usage, please use `TOML` format.
+
+## Minor changes in `TOML` config format
+
+* [`mod_bosh.max_pause`](../../modules/mod_bosh/#modulesmod_boshmax_pause) instead of `maxpause`
+
+* [`mod_caps.cache_life_time`](../../modules/mod_caps/#modulesmod_capscache_life_time): the value should be provided milliseconds
+
+* [`mod_disco.server_info.module`](../../modules/mod_disco/#modulesmod_discoserver_info): the field is optional, no longer required
+
+* [`mod_register`](../../modules/mod_register/#modulesmod_registeriqdisctype), [`mod_vcard`](../../modules/mod_vcard/#modulesmod_vcardiqdisctype) & [`mod_version`](../../modules/mod_version/#modulesmod_versioniqdisctype): `iqdisc.type` default value was changed from `"no_queue"` to `"one_queue"`
+
+* [`mod_event_pusher_push.virtual_pubsub_hosts`](../../modules/mod_event_pusher_push/#modulesmod_event_pusher_pushvirtual_pubsub_hosts): the default value was changed to `[]`
+
+* [`mod_event_pusher_sns.presence_updates_topic`](../../modules/mod_event_pusher_sns/#modulesmod_event_pusher_snspresence_updates_topic): default not set
+
+* `mod_event_pusher_sns.muc_host`: the option was removed
+
+* [`mod_global_distrib.connections.advertised_endpoints`](../../modules/mod_global_distrib/#modulesmod_global_distribconnectionsadvertised_endpoints): default not set (`false` is no longer accepted)
+
+* `mod_global_distrib.connections.tls.enabled`: flag was removed, TLS is enabled by providing [`cacertfile`](../../modules/mod_global_distrib/#modulesmod_global_distribconnectionstlscacertfile) and [`certfile`](../../modules/mod_global_distrib/#modulesmod_global_distribconnectionstlscertfile) options
+
+* [`mod_http_upload.s3`](../../modules/mod_http_upload/#s3-backend-options): the section is mandatory
+
+* [`mod_mam_meta.user_prefs_store`](../../modules/mod_mam/#modulesmod_mam_metauser_prefs_store): `false` value is no longer allowed
```

Comment:
the value 'false' is not longer allowed
alternatively: 'false' is no longer allowed